src/Composer/DependencyResolver/Problem.php

```diff
@@ -241,16 +242,18 @@ public static function getMissingPackageReason(RepositorySet $repositorySet, Req
                 $ext = substr($packageName, 4);
                 $msg = "- Root composer.json requires PHP extension ".$packageName.self::constraintToText($constraint).' but ';
 
-                if (extension_loaded($ext)) {
-                    $version = self::getPlatformPackageVersion($pool, $packageName, phpversion($ext) ?: '0');
-
-                    if (null === $version) {
-                        return array($msg, 'the '.$packageName.' package is disabled by your platform config. Enable it again with "composer config platform.'.$packageName.' --unset".');
+                $version = self::getPlatformPackageVersion($pool, $packageName, phpversion($ext) ?: '0');
+                if (null === $version) {
+                    if (extension_loaded($ext)) {
+                        return array(
+                            $msg,
+                            'the '.$packageName.' package is disabled by your platform config. Enable it again with "composer config platform.'.$packageName.' --unset".',
+                        );
+                    } else {
+                        $error = 'it is missing from your system';
                     }
-
-                    $error = 'it has the wrong version ('.$version.') installed';
                 } else {
-                    $error = 'it is missing from your system';
+                    $error = 'it has the wrong version ('.$version.') installed';
```

Comment:
```suggestion
                    $error = 'it has the wrong version installed ('.$version.')';
```

Hard to read otherwise